File: gatling-performance-suite/src/test/scala/com/example/performance/simulations/SpikeSimulation.scala
package com.example.performance.simulations

import io.gatling.core.Predef._
import io.gatling.http.Predef._
import scala.concurrent.duration._
import com.example.performance.scenarios.UserJourney
import com.example.performance.config.TestConfig

class SpikeSimulation extends Simulation {

  val httpProtocol = http
    .baseUrl(TestConfig.baseUrl)
    .acceptHeader("text/html")
    .userAgentHeader("Mozilla/5.0")

  setUp(
    UserJourney.scenarioDef.inject(
      nothingFor(5.seconds),
      atOnceUsers(100)
    )
  ).protocols(httpProtocol)
   .assertions(
     global.successfulRequests.percent.gt(90),
     global.responseTime.percentile3.lt(4000)
   )
}

File: src/test/java/com/example/performance/simulations/SpikeSimulation.java
package com.example.performance.simulations;

import static java.time.Duration.ofSeconds;

import com.example.performance.config.TestConfig;
import com.example.performance.scenarios.UserJourneyScenario;

import static io.gatling.javaapi.core.CoreDsl.atOnceUsers;
import static io.gatling.javaapi.core.CoreDsl.global;
import static io.gatling.javaapi.core.CoreDsl.nothingFor;
import io.gatling.javaapi.core.Simulation;
import static io.gatling.javaapi.http.HttpDsl.http;
import io.gatling.javaapi.http.HttpProtocolBuilder;

public class SpikeSimulation extends Simulation {

    HttpProtocolBuilder httpProtocol = http
            .baseUrl(TestConfig.BASE_URL)
            .acceptHeader("text/html")
            .userAgentHeader("Mozilla/5.0");

    {
        setUp(
                UserJourneyScenario.scenarioDef.injectOpen(
                        nothingFor(ofSeconds(5)),
                        atOnceUsers(100)
                )
        ).protocols(httpProtocol)
                .assertions(
                        global().successfulRequests().percent().gt(90.0),
                        global().responseTime().percentile3().lt(Integer.valueOf(4000))
                );
    }
}